src/r/recordFactories/ProjectFactory.ts

```diff
@@ -191,6 +172,20 @@ export class ProjectFactory extends RecordFactory<RT.project> {
     }
   }
 
+  /** 
+   * Updated version of addSceneSubRecord
+   * Takes in groupElementId to add record to nth level as well.
+   */
+  addSceneDeepRecord<N extends RT>(this: ProjectFactory, { sceneId, record, position, groupElementId }: { sceneId: number, record: RecordNode<N>, position?: number, groupElementId?: number }): RecordNode<N> | undefined {
+    const sceneJson = this.getRecord(RT.scene, sceneId);
+    if (sceneJson !== undefined) {
+      const addedRecord = (new SceneFactory(sceneJson)).addDeepRecord({record, position, groupElementId});
```

Comment:
addDeepRecord can return `undefined` also. Add a check to handle undefined case also.